Source/Core/Core/DSP/DSPHWInterface.cpp

```diff
@@ -98,7 +98,7 @@ u16 gdsp_mbox_read_l(Mailbox mbx)
   return (u16)value;
 }
 
-void gdsp_ifx_write(u32 addr, u32 val)
+void gdsp_ifx_write(u32 addr, u16 val)
```

Comment:
Since some compilers are broken (*cough* clang *cough*) this needs an appropriate change in our JIT to clear the upper 16 bits of the value when the function is called.